srf_generation/realisation.py

```diff
@@ -0,0 +1,264 @@
+"""
+This module provides classes and functions for working with realisations.
+
+Classes:
+    - FaultJump: Represents a jump across two faults.
+    - RealisationFault: Represents a fault in a realisation.
+    - Realisation: Represents a complete realisation.
+
+Functions:
+    - read_realisation: Reads a realisation from a YAML file.
+    - topologically_sorted_faults: Generates faults in topologically sorted order.
+    - normalise_name: Normalises a name for use in filenames or YAML keys.
+"""
+
+import collections
+import dataclasses
+import re
+from pathlib import Path
+from typing import Generator, Optional, Tuple
+
+import numpy as np
+import qcore.coordinates
+import scipy as sp
+import yaml
+from nshmdb import fault
+from nshmdb.fault import Fault
+
+from srf_generation.source_parameter_generation.common import (
+    DEFAULT_1D_VELOCITY_MODEL_PATH,
+)
+
+
+@dataclasses.dataclass
+class FaultJump:
```

Comment:
Not sure if this is over engineering. I don't think this is actually even being used in the fault jumping code anyways